File: Hayaa.BaseModel/Hayaa.Redis.Client/RedisService.cs
using Hayaa.Common;
using Hayaa.Redis.Client.Config;
using System;
using System.Collections.Generic;
using System.Text;

namespace Hayaa.Redis.Client
{
   public class RedisService
    {
        private RedisFactory redisFactory = null;
        public RedisService(RedisClientConfig config)
        {
            redisFactory = new RedisFactory(config);
        }
        public  void Set<T>(String name, String key, T data) where T:class
        {
            var redis = redisFactory.GetService(name);
            if (redis != null)
            {
                redis.StringSet(key, JsonHelper.SerlaizeSafe<T>(data));
            }
        }
       
        public  T Get<T>(String name, String key) where T : class
        {
            T data = default(T);
            var redis = redisFactory.GetService(name);
            if (redis != null)
            {
              String redisResult=  redis.StringGet(key);
                if (!String.IsNullOrEmpty(redisResult))
                {
                    data = JsonHelper.Deserialize<T>(redisResult);
                }
            }
            return data;
        }
      

        public  void Set<T>(String name, String key, T data,int cacheSeconds) where T : class
        {
            var redis = redisFactory.GetService(name);
            if (redis != null)
            {
                redis.StringSet(key, JsonHelper.SerlaizeSafe<T>(data),new TimeSpan(0,0, cacheSeconds));
            }
        }
        public  void Delete(String name, String key)
        {
          
            var redis = redisFactory.GetService(name);
            if (redis != null)
            {
                redis.KeyDelete(key);
            }
        }
    }
}

File: Java/hayaa.redisclient/src/main/java/hayaa/redis/client/RedisService.java
package hayaa.redis.client;

import org.redisson.api.RBucket;
import org.redisson.api.RKeys;
import org.redisson.api.RedissonClient;

import java.util.concurrent.TimeUnit;

/**
 *
 * @author hsieh
 */
public class RedisService {
    private RedisFactory redisFactory = null;
    public RedisService(RedisClientConfig config)
    {
        redisFactory = new RedisFactory(config);
    }
    public  <T> void set(String name, String key, T data) {
        RedissonClient redis = redisFactory.GetService(name);
        if (redis != null) {
            RBucket<T> bucket = redis.getBucket(key);
            bucket.set(data);
        }
    }
    public  <T> T get(String name, String key) {
        T data = null;
        RedissonClient redis = redisFactory.GetService(name);
        if (redis != null) {
            RBucket<T> bucket = redis.getBucket(key);
            data = bucket.get();
        }
        return data;
    }
    public  <T> void set(String name, String key, T data,int cacheSeconds) {
        RedissonClient redis = redisFactory.GetService(name);
        if (redis != null) {
            RBucket<T> bucket = redis.getBucket(key);
            bucket.set(data,cacheSeconds,TimeUnit.SECONDS);
        }
    }
    public  void delete(String name, String key)
    {
        RedissonClient redis = redisFactory.GetService(name);
        if (redis != null) {
            RKeys keys=redis.getKeys();
            keys.delete(key);
        }
    }
}
